cli/azd/extensions/azure.ai.agents/internal/pkg/agents/agent_yaml/yaml.go

```diff
@@ -81,7 +81,7 @@ type Workflow struct {
 type ContainerAgent struct {
 	AgentDefinition      `json:",inline" yaml:",inline"`
 	Protocols            []ProtocolVersionRecord `json:"protocols" yaml:"protocols"`
-	EnvironmentVariables *[]EnvironmentVariable  `json:"environmentVariables,omitempty" yaml:"environmentVariables,omitempty"`
+	EnvironmentVariables *[]EnvironmentVariable  `json:"environment_variables,omitempty" yaml:"environment_variables,omitempty"`
```

Comment:
I hate snake case in json - any reason to convert to snake case when everything else in the camelCase for `azure.yaml` ?

Reply:
@trangevi any thoughts? I changed the `yaml` tag to `environment_variables` as that's what's used in agent.yaml (it wasn't getting unmarshalled before), and updated the `json` tag to match for consistency